### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by anna-hope**
This is a fork of [word2gauss](https://github.com/seomoz/word2gauss) that has been modified to run on Python 3 (I attempted to preserve compatibility with Python 2.7, but I have only tested the code on Python 3.5, and Python 2 compatibility may have suffered).

In addition to modifying the code to run on Python 3, I have rewritten parts of some functions to be able to withstand certain corner cases. I tried not to break anything in the process, but if you find anything, please let me know.

### CODE REVIEW

**Comment by b4hand** [comment]
It would be really nice to split up the corner case handling from the Python 3 compatibility changes.

Also, it looks like your build is failing.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+This is a fork of [word2gauss](https://github.com/seomoz/word2gauss) that has been modified to run on Python 3 (I attempted to preserve compatibility with Python 2.7, but I have only tested the code on Python 3.5, and Python 2 compatibility may have suffered).
```
Comment (b4hand) [inline README.md]: Assuming we accept this merge, we wouldn't want this change to README.md merged. There would just be a single version of word2gauss which supported both Python 2 and 3.

Comment (lake-hope) [inline README.md]: Yes, this README message can be removed.

File: word2gauss/embeddings.pyx [hunk 2]
```diff
@@ -166,21 +180,21 @@ cdef class GaussianEmbedding:
     cdef gradient_t gradient_func
 
     # pointers to data arrays
-    cdef DTYPE_t* mu_ptr
-    cdef DTYPE_t* sigma_ptr
-    cdef DTYPE_t* acc_grad_mu_ptr
-    cdef DTYPE_t* acc_grad_sigma_ptr
+    cdef DTYPE_t*mu_ptr
```
Comment (b4hand) [inline word2gauss/embeddings.pyx]: This formatting change is both unnecessary for Python 3 support and kind of ugly.

Comment (lake-hope) [inline word2gauss/embeddings.pyx]: I've undone it.

**Comment by anna-hope** [comment]
The build currently seems to be failing due to an ImportError when importing auxlib. There is a similar issue on travis's repo: https://github.com/travis-ci/travis-ci/issues/5814

There appears to be a workaround which involves installing auxlib manually, so I will try that.

**Comment by matt-peters** [comment]
Thanks for the PR!  Just a quick note that I'm on holiday until the end of
May and will review when back in the office.

On Mon, May 2, 2016 at 2:36 PM, Anton Osten notifications@github.com
wrote:

> The build currently seems to be failing due to an ImportError when
> importing auxlib. There is a similar issue on travis's repo:
> travis-ci/travis-ci#5814
> https://github.com/travis-ci/travis-ci/issues/5814
> 
> There appears to be a workaround which involves installing auxlib
> manually, so I will try that.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/seomoz/word2gauss/pull/12#issuecomment-216371769

## 

Dr. Matthew Peters
Data Scientist
Moz

t: @mattthemathman https://twitter.com/mattthemathman

**Comment by anna-hope** [comment]
The Python 3 travis build currently fails because the [vocab module](https://github.com/seomoz/vocab) seems to be incompatible with Python 3. However, the code works correctly with my own implementation of the vocab class.

The Python 2 build fails due to an issue with importing six. I have added the six library to the requirements, but for whatever reason it still can't be imported by travis. I will look into it more and see if I can identify the cause of this problem.

**Comment by matt-peters** [comment]
@notnami : The `vocab` package was only used for testing so I removed it in #13, in addition to adding some numerical gradient checks for the code.  This removes the major hurdle to support Python 3.  Rebase your branch on master, resolve any conflicts and force push then I'll review your PR for merge.  Thanks for your patience!

**Pull request merged**